src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -886,20 +897,24 @@ private static void RestoreViewName(DataSourceEntry ds, string dataSetName)
             var text = PAWriterVisitor.PrettyPrint(ir);
             dir.WriteAllText(subDir, filename, text);
 
-            var extraData = new Dictionary<string, ControlState>();
-            foreach (var item in app._editorStateStore.GetControlsWithTopParent(topParentname ?? controlName))
+            // For TestSuite controls, only the top parent control has an editor state created.
+            // For other control types, create an editor state.
+            if (string.IsNullOrEmpty(topParentName))
             {
-                extraData.Add(item.Name, item);
-            }
+                string editorStateFilename = $"{newControlName}.editorstate.json";
+                EditorStateFile editorStateFile = new EditorStateFile
```

Comment:
Consider moving this constructor after the dictionary is populated. We aren't actually using immutabe data structures here, but it's still generally good practice to not insert into a property of another object directly.